examples/all-clusters-app/all-clusters-common/src/tls-client-management-instance.cpp

```diff
@@ -0,0 +1,168 @@
+        if (item.fabric == fabric)
+        {
+            if (fabricI++ == index)
+            {
+                endpoint = item.payload;
+                return CHIP_NO_ERROR;
+            }
+        }
+    }
+
+    return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+}
+
+ClusterStatusCode TlsClientManagementCommandDelegate::ProvisionEndpoint(
+    EndpointId matterEndpoint, FabricIndex fabric,
+    const TlsClientManagement::Commands::ProvisionEndpoint::DecodableType & provisionReq, uint16_t & endpointID)
+{
+    VerifyOrReturnError(matterEndpoint == EndpointId(1), ClusterStatusCode(Status::ConstraintError));
+
+    if (mCertificateTable.HasRootCertificateEntry(fabric, provisionReq.caid) != CHIP_NO_ERROR)
+    {
+        return ClusterStatusCode::ClusterSpecificFailure(StatusCodeEnum::kRootCertificateNotFound);
+    }
+    if (!provisionReq.ccdid.IsNull() &&
+        mCertificateTable.HasClientCertificateEntry(fabric, provisionReq.ccdid.Value()) != CHIP_NO_ERROR)
+    {
+        return ClusterStatusCode::ClusterSpecificFailure(StatusCodeEnum::kClientCertificateNotFound);
+    }
+
+    // Find existing value to update & check for port/name collisions
+    Provisioned * provisioned = nullptr;
+    for (auto & item : mProvisioned)
+    {
+        const auto & endpointStruct = item.payload;
+        if (!provisionReq.endpointID.IsNull() && endpointStruct.endpointID == provisionReq.endpointID.Value())
+        {
+            provisioned = &item;
+            continue;
+        }
+        if (endpointStruct.hostname.data_equal(provisionReq.hostname) || (endpointStruct.port == provisionReq.port))
+        {
+            return ClusterStatusCode::ClusterSpecificFailure(StatusCodeEnum::kEndpointAlreadyInstalled);
+        }
+    }
+
+    if (provisionReq.endpointID.IsNull())
+    {
+        VerifyOrReturnError(mProvisioned.size() < mTlsClientManagementServer->GetMaxProvisioned(),
+                            ClusterStatusCode(Status::ResourceExhausted));
+
+        provisioned           = &mProvisioned.emplace_back();
+        auto & endpointStruct = provisioned->payload;
+
+        endpointStruct.endpointID = mNextId++;
+        provisioned->fabric       = fabric;
+    }
+    // Updating existing value
+    else if (provisioned == nullptr || provisioned->fabric != fabric)
+    {
+        return ClusterStatusCode(Status::NotFound);
+    }
+
+    auto & endpointStruct = provisioned->payload;
+    ReturnValueOnFailure(endpointStruct.CopyHostnameFrom(provisionReq.hostname), ClusterStatusCode(Status::ResourceExhausted));
+    endpointStruct.port  = provisionReq.port;
+    endpointStruct.caid  = provisionReq.caid;
+    endpointStruct.ccdid = provisionReq.ccdid;
+
+    return ClusterStatusCode(Status::Success);
+}
+
+Status TlsClientManagementCommandDelegate::FindProvisionedEndpointByID(EndpointId matterEndpoint, FabricIndex fabric,
+                                                                       uint16_t endpointID, EndpointStructType & endpoint) const
```

Comment:
Recommend naming endpoint as outEndPoint to make it clear it is an outParam.

Reply:
Done & (& in `GetProvisionedEndpointByID` as well)